lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -508,17 +512,22 @@ def _download_file(url, b_path, expected_hash, validate_certs, token=None, timeo
     return b_file_path
 
 
-def _consume_file(read_from, write_to=None):
+def _consume_file(read_from, write_to=None, length=None):
     # type: (t.BinaryIO, t.BinaryIO) -> str
     bufsize = 65536
     sha256_digest = sha256()
     data = read_from.read(bufsize)
+    actual_length = 0
     while data:
+        actual_length += len(data)
         if write_to is not None:
             write_to.write(data)
             write_to.flush()
         sha256_digest.update(data)
         data = read_from.read(bufsize)
```

Comment:
Perhaps, this should be even
```suggestion
    actual_length = 0
    while data := read_from.read(bufsize):
        actual_length += len(data)
        if write_to is not None:
            write_to.write(data)
            write_to.flush()
        sha256_digest.update(data)
```
?